x/twap/strategy_test.go

```diff
@@ -300,3 +301,21 @@ func (s *TestSuite) TestComputeGeometricStrategyTwap_ThreeAsset() {
 		})
 	}
 }
+
+// TestTwapPow_MaxSpotPrice_NoOverflow tests that no overflow occurs at log_2{max spot price values}.
+// and that the epsilon is within the tolerated multiplicative error.
+func (s *TestSuite) TestTwapLogPow_MaxSpotPrice_NoOverflow() {
+	errTolerance := osmomath.ErrTolerance{
+		MultiplicativeTolerance: sdk.OneDec().Quo(sdk.NewDec(10).Power(18)),
+		RoundingDir:             osmomath.RoundDown,
+	}
+
+	oneYear := OneSec.MulInt64(60 * 60 * 24 * 365)
```

Comment:
Can we test w/ 100 years :eyes:

Reply:
Do we also need to test min spot price, since this can get underflow? (or is that a future PR?)

Reply:
(not blocking merge, as long as we track in an issue otherwise)

Reply:
https://github.com/osmosis-labs/osmosis/issues/3815

Reply:
Tested with 100 years - will address underflow in #3815